### PR TITLE
[Mobile Payments] Remove print and log to Lumberjack where needed

### LINKED ISSUE #5428: [Mobile Payments] [SDK 2] Follow Up: Replace prints with DDLogs
e.g. “error canceling software update” in CardReaderSettingsConnectedViewModel and other places in the card reader code
https://github.com/woocommerce/woocommerce-ios/pull/5322#discussion_r741854612

### PR DESCRIPTION
**Pull request by ctarda**
Closes #5428 

Brought up during the review of #5322 

There are a few print statements left around. Some of them can be completely removed, some others (like error messages) are better logged to Lumberjack.

I went through the Hardware project, and updated or removed all the prints I could find. Looked also into view models, view controllers and actions/stores associated with In-Person Payments.

## Changes
* There are no user-facing changes, or changes in behaviour.
* Removed some usages of `print`
* In some cases, the `print` is completely removed, in others, we will log to `DDLog`

## How to test
* Check that CI is still ✅ 
* Capture a payment. Look at the logs, see if there is anything left behind that should not be there.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift [hunk 1]
```diff
@@ -408,12 +408,12 @@ private extension StripeCardReaderService {
...
 
                     if underlyingError != .commandCancelled {
-                        print("==== collect payment method was not cancelled. this is an actual error ", underlyingError)
+                        DDLogError("💳 Error: collect payment method \(underlyingError)")
                         promise(.failure(CardReaderServiceError.paymentMethodCollection(underlyingError: underlyingError)))
                     }
 
                     if underlyingError == .commandCancelled {
-                        print("==== collect payment method cancelled. this is an error we ignore ", error)
+                        DDLogError("💳 Warning: collect payment error cancelled. We actively ignore this error \(error)")
```
Comment (joshheald) [inline Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift]: I think this should probably be Info or Warn level, given that we ignore it...

**Review by joshheald — APPROVED** [review]
One suggestion

File: Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift (hunk 1 above)
Comment (ctarda) [reply Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift]: Good point! I pushed an update, and enabled auto-merge

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift [hunk 2]
```diff
@@ -411,7 +411,7 @@ private extension CardReaderConnectionController {
                 ServiceLocator.analytics.track(.cardReaderSoftwareUpdateCancelTapped, withProperties: analyticsProperties)
                 cancelable.cancel { result in
                     if case .failure(let error) = result {
-                        print("=== error canceling software update: \(error)")
+                        DDLogError("💳 Error: canceling software update \(error)")
```
Comment (allendav) [inline WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift:414]: Somehow I learned the British spelling. This always catches me when I see it. No change needed, just funny to me :)

https://oneminuteenglish.org/en/canceling-cancelling/

Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift:414]: Welcome to my world. Seeing "typos" everywhere 😉

**Comment by allendav** [comment]
Tests fine. I did have trouble connecting to my Chipper the first time I tested this, and I've had that a couple times recently. It causes the "We couldn’t connect your reader" modal to display and the following was logged:

2021-11-17 09:07:16.144120-0800 WooCommerce[5603:4066408] 🔵 Tracked card_reader_connection_failed, properties: [AnyHashable("error_domain"): "Hardware.CardReaderServiceError", AnyHashable("blog_id"): 197907611, AnyHashable("error_code"): "1", AnyHashable("error_description"): "Hardware.CardReaderServiceError.connection(underlyingError: Hardware.UnderlyingError.bluetoothDisconnected)", AnyHashable("is_wpcom_store"): false]

I'll open a separate issue for that

**Pull request merged**